scripts/tee/CLI/CLI.go

```diff
@@ -0,0 +1,243 @@
+package CLI
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/alixaxel/pagerank"
+	"io/ioutil"
+	"log"
+	"net/http"
+	url2 "net/url"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type CLI struct{}
+
+type Response struct {
+	Blocks   string `json:"blocks"`
+	Duration int    `json:"duration"`
+}
+
+type block struct {
+	TxnContent []content `json:"txn_content"`
+	TxNum      int       `json:"tx_num"`
+}
+
+type content struct {
+	Inputs  []ledger `json:"inputs"`
+	Outputs []ledger `json:"outputs"`
+}
+
+type ledger struct {
+	UserAccount string `json:"userAccount"`
+	Amount      int    `json:"amount"`
+}
+
+type message struct {
+	TxNum      int64    `json:"tx_num"`
+	TxnContent []rawtxn `json:"txn_content"`
+}
+
+type rawtxn struct {
+	Attester uint32  `json:"attester"`
+	Attestee uint32  `json:"attestee"`
+	Score    float64 `json:"score"`
+}
+
+type rawtxnslice []rawtxn
+
+var url = "http://localhost:14700"
+
+func printUsage() {
+
+	fmt.Println("Usage:")
+	fmt.Println("\taddattestationinfo -info <detail> -- add attestation information.")
+	fmt.Println("\tgetrank -num <NUMBER> -period <WHICH_PERIOD> -- get TEE rank of a specific period.")
+	fmt.Println("\tprinthcgraph -period <WHICH_PEIOD> -- output HCGraph information.")
+
+}
+
+func isValidArgs() {
+	if len(os.Args) < 2 {
+		printUsage()
+		os.Exit(1)
+	}
+}
+
+func (cli *CLI) addAttestationInfo(info []string) {
+	raw := new(rawtxn)
+	num, err := strconv.ParseUint(info[0], 10, 64)
+	raw.Attester = uint32(num)
+	num, err = strconv.ParseUint(info[1], 10, 64)
+	raw.Attestee = uint32(num)
+	num, err = strconv.ParseUint(info[2], 10, 64)
+	raw.Score = float64(num)
+	m := new(message)
+	m.TxNum = 1
+	m.TxnContent = []rawtxn{*raw}
+	ms, err := json.Marshal(m)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	data := "{\"command\":\"storeMessage\",\"address\":" + info[0] + ",\"message\":" + url2.QueryEscape(string(ms)) + ",\"tag\":\"TEE\"}"
+	fmt.Println("data : " + data)
+	r := doPost([]byte(data))
+	fmt.Println(r)
+}
+
+func (cli *CLI) getRank(num string, period []string) {
```

Comment:
we need a UT for this function, seriously.

Reply:
cd $GOPATH/src/CLI
go test